### PR TITLE
WT-13806 Fixed incorrect calculation for number of reserved session slots

### PR DESCRIPTION
**Pull request by LX6T**
Wiredtiger calls `__conn_session_size` during `wiredtiger_open` to reserve the session slots that threads can use. Currently for eviction workers and lsm_manager workers, we use the configured max threads value.

Both eviction max worker and lsm_manager max worker can be changed by the **WT_CONNECTION::reconfigure** API, and during the reconfiguration we don't recalculate the number of reserved session slots, causing potential running out of available session slots problem.

For lsm_manager workers I changed the code to use the macro `WT_LSM_MAX_WORKERS` (The max value we can configure for `lsm_manager.worker_thread_max`) to calculate the session slots. I have done the same for eviction, added a `WT_EVICT_MAX_WORKERS` in `evict.h` and used the value to calculate the session slots as well.

### CODE REVIEW

**Comment by sueloverso** [comment]
This looks like the correct change for today's code. @Sean04 just FYI because this will conflict or need to be updated in your LSM-removal branch.

**Review by zunyiliu — APPROVED** [review]
LGTM

File: src/evict/evict.h [hunk 1]
```diff
@@ -137,6 +137,8 @@ struct __wt_evict {
 #define WT_EVICT_CALL_URGENT 0x4u   /* Urgent eviction */
 /* AUTOMATIC FLAG VALUE GENERATION STOP 32 */
 
+#define WT_EVICT_MAX_WORKERS 20
```
Comment (jiechenbo) [inline src/evict/evict.h:140]: Could you add a comment inside api_data.py to note that this number must match the eviciton.threads_max?
E.g.
```
    Config('eviction', '', r'''
        eviction configuration options''',
        type='category', subconfig=[
            Config('threads_max', '8', r'''
                maximum number of threads WiredTiger will start to help evict pages from cache. The
                number of threads started will vary depending on the current eviction load. Each
                eviction worker thread uses a session from the configured session_max''',
                min=1, max=20), # !!! Must match WT_EVICT_MAX_WORKERS
```

**Review by jiechenbo — COMMENTED** [review]
LGTM! please add the comment to api_data.py before merging.

**Review by jiechenbo — APPROVED** [review]

File: src/evict/evict.h (hunk 1 above)
Comment (LX6T) [reply src/evict/evict.h:140]: Ok done!

**Pull request merged**